packages/blockchain-api/src/currencyConversion/CurrencyConversionAPI.ts

```diff
@@ -74,14 +77,25 @@ export default class CurrencyConversionAPI<TContext = any> extends DataSource {
   private getSupportedExchangeRate(
     fromCode: string,
     toCode: string,
-    timestamp?: number
-  ): Promise<BigNumber> {
+    timestamp?: number,
+    impliedCeloToCUSDExchangeRate?: BigNumber.Value
+  ): BigNumber | Promise<BigNumber> {
     const pair = `${fromCode}/${toCode}`
 
     if (pair === 'cUSD/USD' || pair === 'USD/cUSD') {
       // TODO: use real rates once we have the data
-      return Promise.resolve(new BigNumber(1))
-    } else if (pair === 'cGLD/cUSD' || pair === 'cUSD/cGLD') {
+      return new BigNumber(1)
+    } else if (pair === 'cGLD/cUSD') {
+      // If this is an exchange
+      if (impliedCeloToCUSDExchangeRate) {
+        return new BigNumber(impliedCeloToCUSDExchangeRate)
+      }
+      return this.goldExchangeRateAPI.getExchangeRate({
+        sourceCurrencyCode: fromCode,
+        currencyCode: toCode,
+        timestamp,
+      })
+    } else if (pair === 'cUSD/cGLD') {
```

Comment:
Why aren't we doing the same as above for this pair?

Reply:
We are only using `impliedCeloToCUSDExchangeRate` when converting from `cGLD` to `localCurrency` (see here: https://github.com/celo-org/celo-monorepo/blob/c99339c7a9bfafe681fa861f393325cf79af3e4a/packages/blockchain-api/src/schema.ts#L313-L318).